aws/table_aws_elasticache_serverless_cache.go

```diff
@@ -0,0 +1,292 @@
+		return nil, err
+	}
+
+	input := &elasticache.DescribeServerlessCachesInput{
+		MaxResults: aws.Int32(100),
+	}
+
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < *input.MaxResults {
+			if limit < 20 {
+				input.MaxResults = aws.Int32(20)
+			} else {
+				input.MaxResults = aws.Int32(limit)
+			}
+		}
+	}
+
+	// List call
+	paginator := elasticache.NewDescribeServerlessCachesPaginator(svc, input, func(o *elasticache.DescribeServerlessCachesPaginatorOptions) {
+		o.Limit = *input.MaxResults
+		o.StopOnDuplicateToken = true
+	})
+
+	for paginator.HasMorePages() {
+		// apply rate limiting
+		d.WaitForListRateLimit(ctx)
+
+		output, err := paginator.NextPage(ctx)
+		if err != nil {
+			plugin.Logger(ctx).Error("aws_elasticache_serverless_cache.listElastiCacheServerlessCaches", "api_error", err)
+			return nil, err
+		}
+
+		for _, serverlessCache := range output.ServerlessCaches {
+			d.StreamListItem(ctx, serverlessCache)
+
+			// Context can be cancelled due to manual cancellation or the limit has been hit
+			if d.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getElastiCacheServerlessCache(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// Create service
+	svc, err := ElastiCacheClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_elasticache_serverless_cache.getElastiCacheServerlessCache", "connection_error", err)
+		return nil, err
+	}
+
+	serverlessCacheName := d.EqualsQuals["serverless_cache_name"].GetStringValue()
+
+	// Return nil, if no input provided
+	if serverlessCacheName == "" {
+		return nil, nil
+	}
+
+	params := &elasticache.DescribeServerlessCachesInput{
+		ServerlessCacheName: aws.String(serverlessCacheName),
+	}
+
+	op, err := svc.DescribeServerlessCaches(ctx, params)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_elasticache_serverless_cache.getElastiCacheServerlessCache", "api_error", err)
+		return nil, err
+	}
+
+	if len(op.ServerlessCaches) > 0 {
+		return op.ServerlessCaches[0], nil
+	}
+
+	return nil, nil
+}
+
+func listTagsForElastiCacheServerlessCache(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	serverlessCache := h.Item.(types.ServerlessCache)
+
+	// Create session
+	svc, err := ElastiCacheClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_elasticache_serverless_cache.listTagsForElastiCacheServerlessCache", "connection_error", err)
+		return nil, err
+	}
+
+	// Build param
+	param := &elasticache.ListTagsForResourceInput{
+		ResourceName: serverlessCache.ARN,
+	}
+
+	serverlessCacheTags, err := svc.ListTagsForResource(ctx, param)
+
+	if err != nil {
+		var ae smithy.APIError
+		if errors.As(err, &ae) {
+			if ae.ErrorCode() == "ServerlessCacheNotFoundFault" {
+				return nil, nil
+			}
+		}
```

Comment:
I'm not entirely sure if we'll ever encounter the `ServerlessCacheNotFoundFault` error. Since we're not manually providing input, and based on the Steampipe table execution flow, the List/Get hydrate call is always executed first. This ensures that the `ARN` value is available for subsequent hydrate calls. If the List/Get call doesn't return any data, the other hydrate calls won't be triggered.

However, if you've come across a scenario where this error does occur, it would be very helpful to include a comment above the error handling to clarify when and why it happens.

Reply:
Hmm, got it! It’s possible that this is coming from some legacy code that hasn’t been cleaned up yet. Could you please try removing the `ServerlessCacheNotFoundFault` error code handling here and see if that helps? Let me know how it goes!

_Note: After removing the error code, please try running the query and let me know if you still encounter any issues._

Reply:
Hi @michalpl-monday, I was considering the potential race condition and agree that we might encounter the `ServerlessCacheNotFoundFault` in such cases. So yes, I think it’s good to keep the error handling in place. Thanks!